app/controllers/builds_controller.rb

```diff
@@ -92,8 +92,18 @@ def edit_build_params
     params.require(:build).permit(:label, :description)
   end
 
+  def extra_docker_build_params
+    hash = {}
+    Samson::Hooks.fire(:extra_build_params).each do |p|
+      hash[p] = params[p].present?
```

Comment:
This only does the minimum since for this task there is only one extra argument. If there are extra arguments the function can change to suit the need with extra conversions from string values to proper parameter values.